CCompiler/Lexer.cs

```diff
@@ -0,0 +1,164 @@
+namespace CCompiler
+{
+  using System;
+  using System.Collections.Generic;
+  using System.Text;
+  using System.Text.RegularExpressions;
+
+  public class Lexer
+  {
+    [Flags]
+    private enum LexerMode
+    {
+      Normal = 1 << 0,
+
+      Stirng = 1 << 1,
+      StirngEscape = 1 << 2,
+
+      CommentStart = 1 << 3,
+      Comment = 1 << 4,
+      CommentEed = 1 << 5,
+    }
+
+    private static readonly HashSet<char> endWordBreakers = new HashSet<char> { ' ', '\n', '\r' };
+    private static readonly HashSet<char> wordBreakers = new HashSet<char> { ';', '{', '}', '(', ')' };
+
+    private static readonly HashSet<string> Keywords = new HashSet<string>
+    {
+      "int",
+      "return"
+    };
+
+    public List<Token> Lex(string sourceCode)
+    {
+      var ret = new List<Token>();
+      var mode = LexerMode.Normal;
+      var strBuilder = new StringBuilder();
+      int fileIndex = -1;
```

Comment:
Is the first character in a file number 0? or number 1?

Reply:
new Name "lineNumber" with starting value of 0 but there should probably be 1